api/tempo/v1alpha1/tempostack_types.go

```diff
@@ -177,19 +177,32 @@ type MetricsConfigSpec struct {
 // TracingConfigSpec defines a tracing config including endpoints and sampling.
 type TracingConfigSpec struct {
 	// SamplingFraction defines the sampling ratio. Valid values are 0 to 1.
+	// The SamplingFraction has to be defined to enable tracing.
 	//
 	// +optional
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Sampling Fraction"
 	SamplingFraction string `json:"sampling_fraction,omitempty"`
 
 	// JaegerAgentEndpoint defines the jaeger endpoint data gets send to.
+	// Deprecated: in favor of OTLPHttpEndpoint.
 	//
 	// +optional
 	// +kubebuilder:validation:Optional
 	// +kubebuilder:default:="localhost:6831"
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Jaeger-Agent-Endpoint"
 	JaegerAgentEndpoint string `json:"jaeger_agent_endpoint,omitempty"`
+
+	// OTLPHttpEndpoint defines the OTLP/http endpoint data gets send to.
+	// For example, "http://localhost:4320".
+	// The default OTLP/http port 4318 collides with the distributor ports, therefore it is recommended to use a different port
+	// on the sidecar injected to the Tempo (e.g. 4320).
```

Comment:
This can be done in a separate PR, but should we remove the automated sidecar injection as we talked recently? And don't define any `kubebuilder:default` for this setting?

If we want to keep supporting the sidecar injection via the OTEL operator, I'd make it optional (opt-in).

Reply:
I think the the sidecar injection should be optional and controlled via a CR option.

Let's do it in a separate PR https://github.com/grafana/tempo-operator/issues/1111